packages/helix_editor.rb

```diff
@@ -0,0 +1,83 @@
+require 'package'
+
+class Helix_editor < Package
+  description 'A vim inspired editor with LSP support'
+  homepage 'https://helix-editor.com/'
+  version '22.12'
+  license 'MPL-2.0' # license of source
+  compatibility 'all'
+  source_url 'https://github.com/helix-editor/helix/archive/22.12.tar.gz'
+  source_sha256 'edae8af46401b45c3e71c38b4fa99f931c4458127978ccd1b29aaae79331d972' # Use the command "sha256sum"
+
+  depends_on 'rust' => :build
+  depends_on 'xdg_base'
+
+  @no_fhs = true
+  @no_shrink = true
+  @no_strip = true
+  @no_patchelf = true
+
+  @xdg_config_home = ENV.fetch 'XDG_CONFIG_HOME', "#{CREW_PREFIX}/.config"
+  @helix_runtime_dir = "#{@xdg_config_home}/helix"
+  @build_folder_suffix = '/target/release'
+
+  def self.build
+    puts 'Building. This may be long.'
+    system 'cargo build --release --locked'
+  end
+
+  def self.install
+    # Copy executable
+    helix_executable_dest_dir = "#{CREW_DEST_PREFIX}/bin"
+    FileUtils.mkdir_p helix_executable_dest_dir.to_s
+    FileUtils.install ".#{@build_folder_suffix}/hx", helix_executable_dest_dir, mode: 0o755
+    # Copy runtime dir (without the sources)
+    FileUtils.rm_rf './runtime/grammars/sources' # remove the sources
+    helix_runtime_dest_dir = "#{CREW_DEST_DIR}#{@helix_runtime_dir}"
+    FileUtils.mkdir_p helix_runtime_dest_dir
+    FileUtils.cp_r './runtime', helix_runtime_dest_dir
+  end
+
+  def self.check
+    # Ensure hx is executable
+    command_status = system ".#{@build_folder_suffix}/hx --version", exception: false
+    raise 'hx is not executable' unless command_status == true
+  end
+
+  def self.postinstall
+    # This will print a warning if "hx" is not a valid command
+    command_status = system 'hx --version', exception: false
+    puts 'Warning: hx is not in PATH'.lightred unless command_status == true
+    # Check if helix can find its runtime path
+    if command_status == true
+      command_output = `hx --health`
+      unless command_output.include? @helix_runtime_dir
+        puts "Warning: helix cannot find its runtime dir. \
+        The environment variable HELIX_RUNTIME may need to be set.".lightred
+      end
+    end
+    puts <<~EOT2.lightblue
+      Use the 'hx' command to start helix.
+      Use 'hx --health' to see if helix can find its runtime and to see which LSP
+      servers are detected.
+      Note that to be able to load some themes, helix needs to be started in a terminal it recognizes
+      as supporting true colors.
+    EOT2
+  end
+
+  def self.remove
+    # Some files in the directory are write protected...
+    FileUtils.rm_rf @helix_runtime_dir
```

Comment:
Prompt to remove?

Reply:
The user should not modify this folder. If the user wants to override the config, he should do it in `~/.config/helix` instead.

Reply:
Is this safe to delete if the user decides to install the module later?  In other words, will they need to set up and configure the package again if this is removed?

Reply:
If there are no additional files added to this folder then could those be included in the package meta data to be removed during `crew remove helix_editor`?

Reply:
Ok. I get your point. 

I removed this code now. Crew already removes the files itself using the package metadata.

In fact, I added this code because some files in the runtime directory were write protected, and crew failed to remove the runtime folder properly. But these write protected files were in fact all in the source folder (in the `.git database`). Since the sources are not anymore part of the package I can remove this code safely.

Reply:
Should I still promp the user in case crew left some files in `@helix_runtime_dir`?

Reply:
It probably wouldn't hurt to add a prompt in case there is some residue left over that didn't get added to the package files.  If we know that no user data is added there then it would fine to exclude.

Reply:
ok done